### DISCUSSION TITLE
How do I assign different values to variables?

### DISCUSSION THREAD

**Discussion by heyeruoqi** [SilentVoid13/Templater#626] [Templates Showcase]
I would like to implement giving a variable such as {{id}} a different value such as 7117 or 7843 etc. It is too much trouble to enter it manually each time, can the template do this? I can prepare the values of the variables in advance to enable more complex logical operations, e.g. I can easily select these variables by means of drop down menus etc.

**Comment by AB1908** [comment]
```js
<%*
let var = 20;
%>
```

Elsewhere in file:
```
<% var %>
```
should do the trick.

**Comment by welpdx** [comment]
> I would like to implement giving a variable such as {{id}} a different value

When you are using `{{id}}` you are using syntax from the Obsidian Built-in Plugin called [Templates](https://help.obsidian.md/Plugins/Templates). 

If you want to use [this community Templater plugin](https://github.com/SilentVoid13/Templater), use the [Eta templating syntax](https://silentvoid13.github.io/Templater/syntax.html). It might take a bit to get used to, but with some tinkering I'm sure you will get it. 

To set a value to a variable, the example above by @AB1908 an excellent answer.
